### ISSUE TITLE
`expect_attributes_without_reason`

### ISSUE THREAD

**Issue by GoldsteinE** [rust-lang/rust-clippy#13623]
### What it does

The same thing as `allow_attributes_without_reason`, only for `#[expect]`.

### Advantage

I’d like to be able do something like:

```rust
// Allow attributes get stale, you should probably use `#[expect]`.
#![warn(clippy::allow_attributes)]
// And when using `#[expect]` you should specify a reason.
#![warn(clippy::expect_attributes_without_reason)]
```

### Drawbacks

Kinda duplicates `allow_attributes_without_reason`. Maybe it should be a single lint that warns against both.

### Example

```rust
#![expect(clippy::some_lint)]
```

Could be written as:

```rust
#![expect(clippy::some_lint, reason = "false positive")]
```

**Comment by GoldsteinE** [comment]
Oops, sorry, it already works like that, it just needs MSRV.

**Comment by hpenne** [comment]
@GoldsteinE It would be great the change to introduce expect_attributes_without_reason could also change allow_attributes_without_reason so that it only applies or "allow" attributes (not "expect"). "expect" attributes work poorly in some cases (like "dead_code" resulting from "cfg" dependent code), and for those cases "allow" is necessary. I would really find it useful to be able to enforce a policy where "allow" is generally prohibited (use "expect"), but "allow" is allowed if there is a "reason" specified. The extra annoyance at having to write a reason would probably be sufficient to motivate developers to use "expect" whenever possible. This would be possible to set up if "allow_without_reason" ignored "expect".
